Possible fix for ceiling collision

AdamPlenty opened the pull request:
#2376

I'm not sure this is a proper fix, but I couldn't reproduce the bug after this change.

Loobinex commented:
Seems I introduced this issue here: https://github.com/dkfans/keeperfx/pull/1935/files
That PR was introduced to face issues with flying units stuck on hearts, water edges, portals and other 'floor' troubles. Figured I introduced a bug on ceiling there.

AdamPlenty commented:
> Seems I introduced this issue here: https://github.com/dkfans/keeperfx/pull/1935/files That PR was introduced to face issues with flying units stuck on hearts, water edges, portals and other 'floor' troubles. Figured I introduced a bug on ceiling there.

Yes, changing what's in the else bit seems to fix it. I've tried to figure out why creature_cannot_move_directly_to returns false sometimes, but without success. It seems to only occur when both the x and y co-ordinates change, however.

Loobinex commented:
> Yes, changing what's in the else bit seems to fix it. I've tried to figure out why creature_cannot_move_directly_to returns false sometimes, but without success. It seems to only occur when both the x and y co-ordinates change, however.

Point being, now we need to make sure we do not reintroduce bugs where flying creatures get stuck on elevation changes at floor level.
